Gem/Code/Source/SampleComponentManager.cpp

```diff
@@ -486,6 +485,20 @@ namespace AtomSampleViewer
             }
             AZ_Warning("SampleComponentManager", targetSampleFound, "Failed find target sample %s", targetSampleName.c_str());
         }
+        if (commandLine->HasSwitch("timingSamples"))
+        {
+                AZStd::string timingSamplesStr = commandLine->GetSwitchValue("timingSamples", 0);
+                int timingSamplesCount = atoi(timingSamplesStr.c_str());
+                if (timingSamplesCount <= 0 || timingSamplesCount > 1000000)
+                {
+                    timingSamplesCount = static_cast<int>(FrameTimeLogSize);
+                }
```

Comment:
You can reduce this to one line with a clamp function

Reply:
made this change